ksql-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -104,6 +104,18 @@
       KSQL_TABLE_STATESTORE_NAME_SUFFIX_DOC =
       "Suffix for state store names in Tables.";
 
+  public static final String
+      KSQL_QUERY_CLOSE_WAIT_TIME_CONFIG = "ksql.query.close.wait.ms";
+  public static final ConfigDef.Type
+      KSQL_QUERY_CLOSE_WAIT_TIME_TYPE = ConfigDef.Type.LONG;
+  public static final long
+      KSQL_QUERY_CLOSE_WAIT_TIME_DEFAULT = 10000;
+  public static final ConfigDef.Importance
+      KSQL_QUERY_CLOSE_WAIT_TIME_IMPORTANCE = ConfigDef.Importance.MEDIUM;
+  public static final String
+      KSQL_QUERY_CLOSE_WAIT_TIME_DOC =
+      "Max time to wait for KSQL query to close when it is terminated.";
```

Comment:
It would perhaps be more accurate to write 'Max time to wait for KSQL query to be cleaned up after it is terminated'.

Reply:
This only applies to close() method. We need to wait until close is sone before we call cleanUp and internal topic deletion.

Reply:
So what is the semantic supposed to be exactly? The amount of time to wait after the query is terminated before returning?